src/HexEdit.jl

```diff
@@ -1,57 +1,56 @@
 module HexEdit
 
-using Compat
-import Compat.read
+# using Compat
+# import Compat.read
 
-export HexEd, dump!, edit!,
-       find!
+export HexEd, dump!, edit!, find!
 
-type HexEd # :)
+mutable struct HexEd
     _filesize::Int
     _fh::IO
     _offset::UInt64
-
-    function HexEd(filename)
-        _filesize  = filesize(filename)
-        _fh        = open(filename, "r+")
-        _offset    = 0x00
-        new(_filesize, _fh, _offset)
-    end # constructor HexEd
 end # type HexEd
 
+function HexEd(filename::AbstractString)
+    _filesize  = filesize(filename)
+    _fh        = open(filename, "r+")
+    _offset    = 0x00
+    HexEd(_filesize, _fh, _offset)
+end # constructor HexEd
+
 #----------
 # displays data in hex format
 #----------
 function dump_line(s::HexEd, line::Array{UInt8})
     llen = length(line)
     plen = llen % 16
 
-    print("$(hex(s._offset, 8)) ")
+    print("$(uppercase(string(s._offset, base=16, pad=8))) | ")
```

Comment:
Making this uppercase is also a change in behavior. I would recommend proposing changes to the display of output in an issue.